tools/src/main/java/com/nedap/archie/rmobjectvalidator/RMObjectValidator.java

```diff
@@ -191,19 +239,31 @@ private boolean hasNoneWithWrongType(List<RMObjectValidationMessage> subResult)
         return subResult.stream().noneMatch((message) -> message.getType() == RMObjectValidationMessageType.WRONG_TYPE);
     }
 
-    private List<CAttribute> getDefaultAttributeConstraints(CObject cobject, List<CAttribute> attributes) {
+    private List<CAttribute> getDefaultAttributeConstraints(CObject cObject, List<CAttribute> attributes) {
+        List<CAttribute> result = getDefaultAttributeConstraints(cObject.getRmTypeName(), attributes);
+        for(CAttribute attribute:result) {
+           attribute.setParent(cObject);
+        }
+        return result;
+    }
+
+    private List<CAttribute> getDefaultAttributeConstraints(String rmTypeName, List<CAttribute> attributes) {
         List<CAttribute> result = new ArrayList<>();
         HashSet<String> alreadyConstrainedAttributes = new HashSet<>();
         for (CAttribute attribute : attributes) {
             alreadyConstrainedAttributes.add(attribute.getRmAttributeName());
         }
-        RMTypeInfo typeInfo = this.lookup.getTypeInfo(cobject.getRmTypeName());
-        for (RMAttributeInfo defaultAttribute : typeInfo.getAttributes().values()) {
-            if (!defaultAttribute.isComputed()) {
-                if (!alreadyConstrainedAttributes.contains(defaultAttribute.getRmName())) {
-                    CAttribute attribute = constraintImposer.getDefaultAttribute(cobject.getRmTypeName(), defaultAttribute.getRmName());
-                    attribute.setParent(cobject);
-                    result.add(attribute);
+        RMTypeInfo typeInfo = this.lookup.getTypeInfo(rmTypeName);
+        CComplexObject fakeParent = new CComplexObject();
+        fakeParent.setRmTypeName(rmTypeName);
```

Comment:
Maybe do it the other way around? I.e. let `getDefaultAttributeConstraints(String, ...)` call `getDefaultAttributeConstraints(CObject, ...)` with the fakeParent. This prevents creating fakeParent and the double loop in the CObject case.